src/codegen/baseline_jit.cpp

```diff
@@ -0,0 +1,736 @@
+// Copyright (c) 2014-2015 Dropbox, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "codegen/baseline_jit.h"
+
+#include <llvm/ADT/DenseMap.h>
+#include <llvm/ADT/DenseSet.h>
+
+#include "codegen/irgen/hooks.h"
+#include "codegen/memmgr.h"
+#include "core/cfg.h"
+#include "runtime/ics.h"
+#include "runtime/objmodel.h"
+#include "runtime/set.h"
+#include "runtime/types.h"
+
+namespace pyston {
+
+static llvm::DenseSet<CFGBlock*> blocks_aborted;
+
+JitCodeBlock::JitCodeBlock(llvm::StringRef name)
+    : code(malloc(code_size)),
+      entry_offset(0),
+      epilog_offset(0),
+      a((uint8_t*)code, code_size - epilog_size),
+      iscurrently_tracing(false),
+      asm_failed(false) {
+    static StatCounter num_jit_code_blocks("num_baselinejit_code_blocks");
+    num_jit_code_blocks.log();
```

Comment:
I think it's ok that we don't free the memory for now, since we don't free the rest of the codegen structures either :(  Maybe we could just stick another StatCounter here that counts the number of instruction bytes that got allocated?